File: opt-engine/apps/nextday-delivery/src/main/java/dev/aws/proto/apps/nextday/api/response/SolverJob.java
/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

package dev.aws.proto.apps.nextday.api.response;

import dev.aws.proto.core.routing.distance.DistanceMatrix;
import lombok.Data;
import lombok.experimental.SuperBuilder;

@Data
@SuperBuilder
public class SolverJob extends dev.aws.proto.apps.appcore.api.response.DispatchResult {
    /**
     * Execution ID of the step function that triggered the dispatch request.
     */
    private String executionId;

    /**
     * Distance matrix metric information.
     */
    private DistanceMatrix.Metrics distanceMatrixMetrics;

    private String warehouseCode;

    private String warehouseName;

    private String orderDate;

    private int orderCount = -1;
}

File: apps_web/src/models/solver-job.ts
/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: MIT-0
 */

export interface SolverJobData {
  Id: string
  executionId: string
  orderCount: number
  orderDate: string
  score: number
  solverDurationInMs: number
  state: string
  createdAt: number
  warehouseCode: string
  warehouseName: string
}

export const EMPTY_SOLVERJOB_DATA: SolverJobData = {
  Id: '',
  executionId: '',
  orderCount: 0,
  orderDate: '',
  score: 0,
  solverDurationInMs: 0,
  state: '',
  createdAt: Date.now(),
  warehouseCode: '',
  warehouseName: '',
}
